tools/generate_events.py

```diff
@@ -212,21 +226,29 @@ def validate_property_types_and_return_extra_imports(e):
 
 def validate_event(e):
     REQUIRED_CLUSTER_PROPERTIES = ["cluster_id"]
-    REQUIRED_HOST_PROPERTIES = ["cluster_id", "host_id"]
-    INVALID_CLUSTER_PROPERTIES = ["host_id"]
+    REQUIRED_HOST_PROPERTIES = ["host_id", "infra_env_id"]
+    REQUIRED_INFRA_ENV_PROPERTIES = ["infra_env_id"]
+    INVALID_CLUSTER_PROPERTIES = ["host_id", "infra_env_id"]
     INVALID_HOST_PROPERTIES = []
+    INVALID_INFRA_ENV_PROPERTIES = ["cluster_id", "host_id"]
```

Comment:
according to the definition of the infraenv type, it is possible to have cluster-id (optionally).

Reply:
done